test/flatten_test.js

```diff
@@ -0,0 +1,80 @@
+goog.module('datalayerhelper.helper.testing.flatten');
+goog.setTestOnly();
+
+const {DataLayerHelper} = goog.require('helper');
+
+describe(`The 'flatten' method of helper`, () => {
+  beforeEach(function() {
+    this.dataLayer = [{a: 1, b: {c: {d: 4}, e: 5}}];
+    this.dataLayer.push({f: 6});
+    this.dataLayer.push({g: 7});
+
+    this.h = new DataLayerHelper(this.dataLayer);
```

Comment:
This name should probably be a full word, e.g. this.helper, rather than a single character.

Style guide recommends: "Do not use abbreviations that are ambiguous or unfamiliar to readers outside your project" (basically just an exception for loop indexes and other common patterns).

https://google.github.io/styleguide/jsguide.html#naming-rules-common-to-all-identifiers

Reply:
Right, fixed